str2host.c

```diff
@@ -344,7 +344,7 @@ ldns_str2rdf_dname(ldns_rdf **d, const char *str)
 	pq = buf;
 	label_len = 0;
 	for (s = str; *s; s++, q++) {
-		if (q > buf + LDNS_MAX_DOMAINLEN) {
+		if (q >= buf + LDNS_MAX_DOMAINLEN) {
```

Comment:
Op regel 317 staat `uint8_t buf[LDNS_MAX_DOMAINLEN + 1];`
dus dan is `q == buf + LDNS_MAX_DOMAINLEN] ` toch prima verder?

Reply:
(for future reference, as this is already merged)

`q == buf + LDNS_MAX_DOMAINLEN` indeed won't overflow the `uint8_t buf[LDNS_MAX_DOMAINLEN + 1];`. But returning a 256 byte long domain name does not make sense. So `q` should be smaller than `buf + LDNS_MAX_DOMAINLEN` and buffer size can be reduced to LDNS_MAX_DOMAINLEN.